rust/src/faucet.rs

```diff
@@ -0,0 +1,35 @@
+use crate::config::maker_endpoint;
+use anyhow::anyhow;
+use anyhow::bail;
+use anyhow::Result;
+use reqwest::StatusCode;
+
+pub async fn call_faucet(address: String) -> Result<Option<String>> {
+    let client = reqwest::Client::builder()
+        .timeout(crate::config::TCP_TIMEOUT)
+        .build()?;
+    let result = client
+        .get(maker_endpoint() + "/api/faucet/" + address.as_str())
+        .send()
+        .await;
+    let response = match result {
+        Ok(res) => res,
+        Err(err) => {
+            tracing::error!("Could not call faucet {err:?}");
+            return Ok(None);
+        }
```

Comment:
❓ Why not return an error here? What's the `Ok(None)` case for?

Reply:
Fair point, fixed it.